ssl/statem/extensions_clnt.c

```diff
@@ -314,7 +314,8 @@ EXT_RETURN tls_construct_ctos_session_ticket(SSL_CONNECTION *s, WPACKET *pkt,
         return EXT_RETURN_NOT_SENT;
 
     if (!WPACKET_put_bytes_u16(pkt, TLSEXT_TYPE_session_ticket)
-            || !WPACKET_sub_memcpy_u16(pkt, s->session->ext.tick, ticklen)) {
+            || (s->session != NULL 
+            && !WPACKET_sub_memcpy_u16(pkt, s->session->ext.tick, ticklen))) {
```

Comment:
This would be clearly wrong. Not sure if it can at all happen that `s->session` would be NULL at this point but if anything `!WPACKET_sub_memcpy_u16(pkt, s->session != NULL ? s->session->ext.tick : NULL , ticklen)` would be more correct as `ticklen` will be definitely 0 if `s->session` is NULL. So this will at least output an empty subpacket rather than nothing.

@mattcaswell might perhaps know if s->session can be NULL at all here and if it would not be better to just return EXT_RETURN_FAIL if it is NULL.